tests/kernel/wave/attention/quantized_attention_test.py

```diff
@@ -0,0 +1,149 @@
+        num_kv_heads=input_shape[0],
+        query_seq_len=input_shape[1],
+        head_size_kv=input_shape[2],
+        head_size=input_shape[3],
+        kv_seq_len=input_shape[4],
+    )
+    q_scale = 0.02578124962747097
+    k_scale = 0.02363281324505806
+    v_scale = 0.010286458767950535
+    (
+        base_attention,
+        hyperparams,
+        dynamic_symbols,
+        dynamic_symbols_map,
+    ) = get_brevitas_pertensor_fp8_attention_kernel(
+        shape, mfma_variant, q_scale=q_scale, k_scale=k_scale, v_scale=v_scale
+    )
+    q_shape = (shape.num_query_heads, shape.query_seq_len, shape.head_size)
+    k_shape = (shape.num_kv_heads, shape.kv_seq_len, shape.head_size)
+    v_shape = (shape.num_kv_heads, shape.kv_seq_len, shape.head_size_kv)
+    o_shape = (shape.num_query_heads, shape.query_seq_len, shape.head_size_kv)
+    hyperparams.update(get_default_scheduling_params())
+
+    perf_filename = request.node.name + ".json"
+    options = WaveCompileOptions(
+        subs=hyperparams,
+        schedule=enable_scheduling,
+        use_scheduling_barriers=enable_scheduling_barriers,
+        dynamic_symbols=dynamic_symbols,
+        dynamic_symbols_map=dynamic_symbols_map,
+        run_bench=run_bench,
+        waves_per_eu=2,
+        denorm_fp_math_f32="preserve-sign",
+        benchmark_batch_size=10,
+        benchmark_repetitions=3,
+        benchmark_results_file=(
+            os.path.join(dump_perf, "tk_" + perf_filename) if dump_perf else None
+        ),
+    )
+
+    options = set_default_run_config(options)
+    base_attention = wave_compile(options, base_attention)
+
+    torch.manual_seed(0)
+    # Smaller range to help with FP8 minimum range
+    MAX_RANGE = 128
+
+    q = torch.ceil(
+        torch.clamp(
+            device_randn(q_shape, dtype=torch.float16) * MAX_RANGE,
+            -MAX_RANGE,
+            MAX_RANGE,
+        )
+    )
+    k = torch.ceil(
+        torch.clamp(
+            device_randn(k_shape, dtype=torch.float16) * MAX_RANGE,
+            -MAX_RANGE,
+            MAX_RANGE,
+        )
+    )
+    v = torch.ceil(
+        torch.clamp(
+            device_randn(v_shape, dtype=torch.float16) * MAX_RANGE,
+            -MAX_RANGE,
+            MAX_RANGE,
+        )
+    )
+    output = device_zeros(o_shape, dtype=torch.float32)
+    asm = base_attention(q, k, v, output)
+    torch_ref = torch.nn.functional.scaled_dot_product_attention(
+        q.to(torch.float32) * q_scale,
+        k.to(torch.float32) * k_scale,
+        v.to(torch.float32) * v_scale,
+    )
+
+    if dump_generated_mlir:
+        filename = f"wave_attention_{'x'.join(map(str, input_shape))}.mlir"
+        with open(filename, "w") as f:
+            f.write(asm)
+
+    rmse = torch.sqrt(torch.mean(torch.square(output - torch_ref)))
+    # Higher tolerance because, we are testing a higher range
+    # of numbers here (-128, 128), typically device_rand just gets (-1, 1).
+    assert rmse < 0.04
```

Comment:
What should the acceptable tolerance be and how can we calculate it as a function of the range of numbers? Is there such an equation? accuracy_tol = 1e-3 * range_max ?

Reply:
will discuss with Giuseppe and add in next PR, but fwiw if I do regular `device_randn` we get rmse of 1e-6 which is much better than `0.006` of pure FP8 attention